roles/docker_private_registry/tasks/main.yaml

```diff
@@ -0,0 +1,38 @@
+---
+#
+#  This role sets up and creates an insecure private registry on the localhost
+#  at port 5000. 
+#
+  - name: Fail if ansible_docker0.ipv4.address is not defined
+    fail: 
+      msg: "ansible_docker0.ipv4.address it not defined!"
+    when: ansible_docker0.ipv4.address is not defined
+    run_once: true
+
+  - name: edit /etc/sysconfig/docker file to add private registry
+    replace: 
+      dest=/etc/sysconfig/docker 
+      regexp='^ADD_REGISTRY=\'--add-registry registry.access.redhat.com\'' 
```

Comment:
I was trying to make the documentation example work on Fedora and CentOS  Since we use containerized kube (apiserver, controller-mgr, and scheduler), I pulled these container images from the Red Hat registry.